### ISSUE TITLE
Alternative to vue-resource?

### ISSUE THREAD

**Issue by garethmcc** [vue-bulma/vue-admin#141]
I ask this because vue-resource has essentially been abandoned:
https://medium.com/the-vue-point/retiring-vue-resource-871a82880af4#.oem7a4wx7

**Comment by luventa** [comment]
Thanks man.
I think these are two approaches:
1. keep using vue-resource since it still works. And wait for the next official recommendation of ajax library.
2. start to use other popular libraries like Axios, which is mentioned in the essay. (I think this choice it the better one)

I will discuss with Fundon and try to make some changes on vue-admin for this.

Thanks again for raising this issue!

**Comment by fundon** [comment]
Axios +1

**Comment by garethmcc** [comment]
According to the article I linked to, Vue does not plan to ever have a recommended ajax library. They want to move away from that completely as it is not core focus for Vue, which in all honesty makes 100% sense from their point of view.

**Comment by luventa** [comment]
@garethmcc vue-resource was kind like an official recommended ajax plugin since vue version 1.x. [Existing Plugins & Tools](http://v1.vuejs.org/guide/plugins.html#Existing-Plugins-amp-Tools)   

We understood Evan's point that Vue.js shall not focus on other functionalities. But related plugins can be found in the documents. Vue2's documents seem never recommend any ajax plugin but I guess they will get one :D

**Comment by dudesweet100** [comment]
Axios +1. Tried to use it, but proxyTable is ignored :(

**Comment by mirari** [comment]
It seems like the only affected plugin in vue-admin is vue-nprogress, I tryed to modify it like this:
```
        const axios = applyOnHttp && Vue.axios
        if (axios) {
          axios.interceptors.request.use(function (request) {
            if (!'showProgressBar' in request) request.showProgressBar = applyOnHttp
            if (request.showProgressBar) initProgress()
            return request;
          }, function (error) {
            return Promise.reject(error);
          });

          axios.interceptors.response.use(function (response) {
            if (response.config.showProgressBar) increase()
            return response
          }, function (error) {
            if (error.config && error.config.showProgressBar) increase()
            return Promise.reject(error);
          });
        }
```

**Comment by luventa** [comment]
I think there is no need to make any change with vue-nprogress. Will commit and merge my changes for this issue soon.

**Comment by luventa** [comment]
My apologies @mirar, after tried some approaches, I decided to add your codes.
At first, I planed to export two methods `start` and 'done', to keep vue-progress isolated. 
But then I found it requires to add codes in `app.js`, which is not so convenient for using this progress component...

Thanks for your sharing!

**Comment by luventa** [comment]
Axios demo is landed in commit [9ac66e3](https://github.com/vue-bulma/vue-admin/commit/9ac66e3d77c01197ddadfb94b8c3c0f7a0bf7489) ; )

**Comment by mirari** [comment]
Thanks for the demo
